### PR TITLE
feat: robust cross-referencing with MyST, cleanup examples

### PR DESCRIPTION
**Pull request by CalMacCQ**
# Description

Cleanup of all of the sphinx crossrefernces before the migration to docs.quantinuum.com/tket happens.

See this useful page on MyST crosslinking syntax -> https://myst-parser.readthedocs.io/en/latest/syntax/cross-referencing.html

I've also added a link checker to the C.I. --> [1e2c271](https://github.com/CQCL/pytket-docs/pull/392/commits/1e2c2716ce0da88d6c0b020d781c2141ca18c1d7) and tested that it works it with a legacy example.

Speaking of legacy examples I've also deleted `Forest_portabiltity_example` and `Spam_example` as I think they are too outdated to be of any use.

I've tried to avoid using full URLs wherever possible and use intersphinx to link to all of the extension packages.
This means we should only need to change URLs in `conf.py`.

# Related issues

Please mention any github issues addressed by this PR.

### CODE REVIEW

File: docs/manual/manual_circuit.md [hunk 1]
```diff
@@ -727,7 +727,7 @@ Despite the {py:class}`~pytket.circuit.Circuit` class having methods for adding
 ```
 
 When constructing subroutines to implement quantum algorithms it is natural to distinguish different groups of qubits. For instance, in the quantum phase estimation algorithm (QPE) we would want to distinguish between state preparation qubits and ancillary qubits which are measured to yield an approximation of the phase.
-The QPE can then be used as a subroutine in other algorithms: for example, integer factoring or estimating the ground state energy of some molecule. For more on the phase estimation algorithm see the [QPE example notebook](https://tket.quantinuum.com/user-guide/examples/algorithms_and_protocols/phase_estimation.html).
```
Comment (CalMacCQ) [inline docs/manual/manual_circuit.md:730]: This refernce isn't working. need to fix

File: docs/manual/manual_circuit.md [hunk 2]
```diff
@@ -1012,7 +1012,7 @@ These Pauli gadget circuits have interesting algebraic properties which are usef
 
 A Pauli gadget can be expressed as $V \, A \, V^\dagger$ where $V$ is the the circuit composed of CX gates and single qubit basis rotations on the right hand side of the Rz gate and $A$ is the Rz gate itself. This observation allows one to construct controlled Pauli gadgets much more efficently. See the [blog post](https://tket.quantinuum.com/blog/posts/controlled_gates/) on the {py:class}`~pytket.circuit.ConjugationBox` construct for more details.
 
-For further discussion see the research publication on phase gadget synthesis [^cite_cowt2020]. Ideas from this paper are implemented in TKET as the [OptimisePhaseGadgets](https://tket.quantinuum.com/api-docs/passes.html#pytket.passes.OptimisePhaseGadgets) and [PauliSimp](https://tket.quantinuum.com/api-docs/passes.html#pytket.passes.PauliSimp) optimisation passes.
+For further discussion see the research publication on phase gadget synthesis [^cite_cowt2020]. Ideas from this paper are implemented in TKET as the OptimisePhaseGadgets(inv:#.~passes.OptimisePhaseGadgets) and PauliSimp (inv:#*.PauliSimp) optimisation passes.
```
Comment (CalMacCQ) [inline docs/manual/manual_circuit.md:1015]: These aren't working either as there are multiple valid import paths

File: docs/manual/manual_circuit.md [hunk 3]
```diff
@@ -1456,11 +1456,11 @@ print(circ.is_symbolic())   # returns True when free_symbols() is non-empty
...
+There are some minor drawbacks associated with symbolic compilation. When using [Euler-angle equations](inv:#*.EulerAngleReduction) or quaternions for merging adjacent rotation gates, the resulting angles are given by some lengthy trigonometric expressions which cannot be evaluated down to just a number when one of the original angles was parameterised; this can lead to unhelpfully long expressions for the angles of some gates in the compiled circuit. It is also not possible to apply the {py:class}`pytket.passes.KAKDecomposition` pass to simplify a parameterised circuit, so that pass will only apply to non-parameterised subcircuits, potentially missing some valid opportunities for optimisation.
 ```
 
 ```{seealso}
-To see how to use symbolic compilation in a variational experiment, have a look at our [VQE (UCCSD) example](https://tket.quantinuum.com/user-guide/examples/algorithms_and_protocols/ucc_vqe.html).
```
Comment (CalMacCQ) [inline docs/manual/manual_circuit.md:1463]: need to fix

File: docs/manual/manual_circuit.md (hunk 1 above)
Comment (CalMacCQ) [reply docs/manual/manual_circuit.md:730]: done in https://github.com/CQCL/pytket-docs/pull/392/commits/875f6c79f87942040797139ad5474c48867c04f4

File: docs/manual/manual_circuit.md (hunk 3 above)
Comment (CalMacCQ) [reply docs/manual/manual_circuit.md:1463]: done in https://github.com/CQCL/pytket-docs/pull/392/commits/875f6c79f87942040797139ad5474c48867c04f4

File: docs/manual/manual_circuit.md (hunk 2 above)
Comment (CalMacCQ) [reply docs/manual/manual_circuit.md]: done in [c3e1312](https://github.com/CQCL/pytket-docs/pull/392/commits/c3e1312b41b9231a228c48e7b8f9a54b51267fba)

**Comment by cqc-melf** [comment]
Are there plans to move the qujaxs docs to docs.quantinuum.com?

**Comment by CalMacCQ** [comment]
> Are there plans to move the qujaxs docs to docs.quantinuum.com?

The pytket-qujax docs are already there -> https://tket.quantinuum.com/extensions/pytket-qujax/

As for the main qujax docs I don't think moving them to the docs.quantinuum.com site is a priority. They're fine on cqcl.github.io for now

File: docs/examples/backends/Forest_portability_example.ipynb
[the remark this answers is not in the document]
Comment (CalMacCQ) [reply docs/examples/backends/Forest_portability_example.ipynb:1]: I removed `Forest_portability_example` I think it is not helpful to have that.

I think we should have a new example for pytket-pyquil

File: README.md [hunk 4]
```diff
@@ -21,7 +21,7 @@ To install, run:
 pip install pytket
 ```
 
-API documentation can be found at https://tket.quantinuum.com/api-docs.
+API documentation can be found at https://docs.quantinuum.com/tket/api-docs.
```
[the remark this answers is not in the document]
Comment (CalMacCQ) [reply README.md:24]: Yes, I would like to change that for the extensions as soon as we can.

**Comment by cqc-melf** [comment]
> > Are there plans to move the qujaxs docs to docs.quantinuum.com?
> 
> The pytket-qujax docs are already there -> https://tket.quantinuum.com/extensions/pytket-qujax/
> 
> As for the main qujax docs I don't think moving them to the docs.quantinuum.com site is a priority.

I was talking about this: https://cqcl.github.io/qujax/

**Review by cqc-melf — APPROVED** [review]

**Pull request merged**